src/EntryPoints/WikibaseFacetedSearchHooks.php

```diff
@@ -26,11 +35,94 @@ public static function onShowSearchHitTitle(
 		array &$query,
 		array &$attributes
 	): void {
-		if ( $title->getNamespace() !== WB_NS_ITEM ) {
+		$itemId = self::getItemId( $title );
+
+		if ( $itemId === null ) {
+			return;
+		}
+
+		$pageTitle = self::getItemPage( $itemId );
+
+		if ( $pageTitle === null ) {
 			return;
 		}
 
-		// TODO: get item site link and replace $title
+		$title = $pageTitle;
+
+		if ( !( $result instanceof EntityResult ) ) {
+			self::rewriteLinkForNonEntityResult(
+				self::newLabelDescriptionLookup( $specialSearch->getContext() ),
+				self::newLinkFormatter( $specialSearch->getLanguage() ),
+				$itemId,
+				$titleSnippet,
+				$attributes
+			);
+		}
+	}
+
+	private static function getItemId( Title $title ): ?ItemId {
+		$entityId = WikibaseRepo::getEntityIdLookup()->getEntityIdForTitle( $title );
+
+		if ( $entityId instanceof ItemId ) {
+			return $entityId;
+		}
+
+		return null;
+	}
+
+	private static function getItemPage( ItemId $itemId ): ?Title {
+		return WikibaseFacetedSearchExtension::getInstance()->getItemPageLookup()->getPageTitle( $itemId );
+	}
+
+	private static function newLinkFormatter( Language $language ): EntityLinkFormatter {
+		return WikibaseRepo::getEntityLinkFormatterFactory()->getDefaultLinkFormatter( $language );
+	}
+
+	private static function newLabelDescriptionLookup( IContextSource $context ): LanguageFallbackLabelDescriptionLookup {
+		return new LanguageFallbackLabelDescriptionLookup(
+			WikibaseRepo::getTermLookup(),
+			WikibaseRepo::getLanguageFallbackChainFactory()->newFromContext( $context )
+		);
+	}
+
+	/**
+	 * @param string[] $attributes
+	 */
+	private static function rewriteLinkForNonEntityResult(
+		LanguageFallbackLabelDescriptionLookup $labelDescriptionLookup,
+		EntityLinkFormatter $linkFormatter,
+		ItemId $itemId,
+		string|HtmlArmor|null &$titleSnippet,
+		array &$attributes
+	): void {
+		$labelData = self::termFallbackToTermData(
+			$labelDescriptionLookup->getLabel( $itemId )
+		);
+		$descriptionData = self::termFallbackToTermData(
+			$labelDescriptionLookup->getDescription( $itemId )
+		);
+
+		$titleSnippet = new HtmlArmor( $linkFormatter->getHtml( $itemId, $labelData ) );
+
+		$attributes['title'] = $linkFormatter->getTitleAttribute(
+			$itemId,
+			$labelData,
+			$descriptionData
+		);
+	}
+
+	/**
+	 * @return string[]|null
+	 */
+	private static function termFallbackToTermData( ?TermFallback $term = null ): ?array {
+		if ( $term ) {
+			return [
+				'value' => $term->getText(),
+				'language' => $term->getActualLanguageCode(),
+			];
+		}
+
+		return null;
```

Comment:
There is a lot of static code here. Likely we can move much to one or more classes that do not bind to globals and can be tested easier